design-proposals/amd-sev.md

```diff
@@ -0,0 +1,245 @@
+    <dhCert>RBBBSDDD=FDDCCCDDDG</dhCert>
+    <session>AAACCCDD=FFFCCCDSDS</session>
+  </launchSecurity>
+  ...
+</domain>
+```
+
+To start a SEV VM there is a need to specify `cbitpos` and `reducedPhysBits`. Those parameters are hypervisor-dependent and can be obtained from the `<sev></sev>` element of the `virsh domcapabilities`:
+
+```xml
+<domainCapabilities>
+...
+  <features>
+    ...
+    <sev supported='yes'>
+      <cbitpos>47</cbitpos>
+      <reducedPhysBits>1</reducedPhysBits>
+    </sev>
+    ...
+  </features>
+</domainCapabilities>
+```
+
+The `policy` parameter allows setting the guest policy flags. Additionally the optional `dhCert` and `session` provide the guest owner's base64 encoded `DH` (Diffie-Hellman) key and the guest owner's base64 encoded `launch blob` respectively.
+
+There are also several prerequisites to run SEV guests:
+
+- All virtio devices need to be configured with the `iommu='on'` attribute in their `<driver>` configuration
+- All memory regions used by the VM must be `locked` for Direct Memory Access (DMA) and to prevent swapping
+
+## Attestation
+
+[Attestation](https://github.com/AMDESE/sev-tool#proposed-provisioning-steps) is an interactive process between platform owner (`host`) and guest owner (`user`). It is carried out to establish a certain level of trust. To leverage the benefits of memory encryption a guest owner provides an encrypted VM disk. To decrypt the data and to run a VM with such a disk there is a need to `inject a secret` into the VM when the system is about to boot. The secret is known only by the guest owner. The goal of the attestation process is to validate that the platform is genuine and it really runs a non-compromised SEV hardware. Also it is needed to establish a private communication channel with the platform so the guest owner can securely provide the secret. Attestation consists of the following steps:
+
+### Step #1 Validation of the platform
+
+The guest owner should get the assets listed bellow from the platform owner:
+
+- the `Platform Diffie-Hellman (PDH)` key exported from the SEV hardware
+- the complete key chain (up to the `root of trust`) used to sign the `PDH`
+- the OVMF image
+
+On a node where KubeVirt is running, the privileged `virt-handler` pod can export the required assets in advance to a `kubevirt-private` directory. Whenever a new VM requiring SEV is launched, the directory can be mounted to the `virt-launcher` pod in `read-only` mode. A user can then invoke the `virtctl` tool to download the certificate chain for validation (for that an additional command will be needed, e.g. `virtctl fetchcertchain ...`).
+
+In order to ensure that the platform is not compromised the guest owner needs to validate the `PDH` key using the provided chain. For that `sev-tool --validate_cert_chain ...` can be used (AMD root certificate should be downloaded from the web for verification). Additionally the UEFI image should be considered 'trustable' by the guest owner.
```

Comment:
Where should this be done, in the guest OS?

Reply:
This is the 'preparation' step and it is done outside of the gest OS. Here the guest owner takes the certificates extracted from the platform and verifies that they are properly signed.

Reply:
> With other words users do something like this?

Correct.